internal/dag/httpproxy_processor.go

```diff
@@ -304,11 +302,16 @@ func (p *HTTPProxyProcessor) computeHTTPProxy(proxy *contour_api_v1.HTTPProxy) {
 				}
 				if tls.ClientValidation.CACertificate != "" {
 					secretName := k8s.NamespacedNameFrom(tls.ClientValidation.CACertificate, k8s.DefaultNamespace(proxy.Namespace))
-					cacert, err := p.source.LookupCASecret(secretName)
+					cacert, err := p.source.LookupCASecret(secretName, proxy.Namespace)
 					if err != nil {
-						// PeerValidationContext is requested, but cert is missing or not configured.
-						validCond.AddErrorf(contour_api_v1.ConditionTypeTLSError, "ClientValidationInvalid",
-							"Spec.VirtualHost.TLS client validation is invalid: invalid CA Secret %q: %s", secretName, err)
+						if _, ok := err.(DelegationNotPermittedError); ok {
+							validCond.AddErrorf(contour_api_v1.ConditionTypeTLSError, "DelegationNotPermitted",
+								"Spec.VirtualHost.TLS CA Secret %q is invalid: %s", tls.ClientValidation.CACertificate, err)
```

Comment:
It'd be ideal to add some test coverage for this (valid delegation, invalid/missing delegation)

Reply:
Done!